File: src/Backend/InventarioEscolar.Domain/Entities/Asset.cs
using InventarioEscolar.Domain.Enums;
using InventarioEscolar.Domain.Interfaces;

namespace InventarioEscolar.Domain.Entities
{
    public class Asset : EntityTenantBase
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public long? PatrimonyCode { get; set; } 
        public decimal? AcquisitionValue { get; set; }
        public ConservationState ConservationState { get; set; }
        public string? SerieNumber { get; set; }
        public long CategoryId { get; set; }
        public Category Category { get; set; } = null!;
        public long? RoomLocationId { get; set; }
        public RoomLocation? RoomLocation { get; set; }

    }
}

File: frontend/EscolarInventarioApp/src/app/core/models/Asset.ts
import { BaseEntity } from "./BaseEntity";
import { Category } from './Category';
import { ConservationState } from "./enum/ConservationState.enum";
import { RoomLocation } from './RoomLocation';

export interface Asset extends BaseEntity {
  name: string;
  description?: string;
  patrimonyCode?: number;
  acquisitionValue?: number;
  conservationState: ConservationState;
  serieNumber?: string;
  category: Category;
  roomLocation?: RoomLocation;
}
